packages/salesforcedx-vscode-soql/src/sfdx/index.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Connection } from '@salesforce/core';
+import { ChannelService } from '@salesforce/salesforcedx-utils-vscode/out/src/commands';
+import { DescribeSObjectResult } from 'jsforce';
+import * as vscode from 'vscode';
+import { nls } from '../messages';
+
+export const channelService = ChannelService.getInstance(
+  nls.localize('soql_channel_name')
+);
+
+const sfdxCoreExtension = vscode.extensions.getExtension(
+  'salesforce.salesforcedx-vscode-core'
+);
+const sfdxCoreExports = sfdxCoreExtension
+  ? sfdxCoreExtension.exports
+  : undefined;
```

Comment:
Why is this necessary?  Wouldn't this be fatal?

Reply:
Good point. I simply moved the existing code without giving it much thought.

I'll add an assertion/safety check. The Extension should fail at activation time if that's `undefined`.

Reply:
done